src/com/facebook/buck/go/GoDescriptors.java

```diff
@@ -142,6 +153,29 @@ static GoCompile createGoCompileRule(
         platform);
   }
 
+  @VisibleForTesting
+  static ImmutableMap<Path, Path> getPackageImportMap(
+      Path basePackageName, Iterable<Path> packageNameIter) {
+    Map<Path, Path> importMapBuilder = Maps.newHashMap();
+    ImmutableSortedSet<Path> packageNames = ImmutableSortedSet.copyOf(packageNameIter);
+
+    Path prefix = Paths.get("");
+    for (Path component : FluentIterable.of(new Path[]{Paths.get("")}).append(basePackageName)) {
+      prefix = prefix.resolve(component);
+
+      Path vendorPrefix = prefix.resolve("vendor");
+      for (Path vendoredPackage : packageNames.tailSet(vendorPrefix)) {
+        if (!vendoredPackage.startsWith(vendorPrefix)) {
+          continue;
```

Comment:
It's not quite the prefix. We're iterating through prefixes, but we're checking for `<prefix>/vendor` - i.e. for a package in `a/b/c` we'll check (& overwrite - longest wins here): `vendor/`, `a/vendor/`, `a/b/vendor/`, `a/b/c/vendor/`.